firedrake/mesh.py

```diff
@@ -224,6 +224,36 @@ def facet_cell_map(self):
                        "facet_to_cell_map")
 
 
+@PETSc.Log.EventDecorator()
+def _from_netgen(ngmesh,comm=None):
+    """
+    Create a DMPlex from an Netgen mesh
+
+    :arg ngmesh: NetGen Mesh
+    """
+    if len(ngmesh.Elements3D()) != 0:
+        raise RuntimeError("Only 2D meshes are implemented at the moment.")
+    else:
+        V = [];
+        T = [];
+        for v in list(ngmesh.Points()):
+            V = V + [[float(v[0].__repr__()),float(v[1].__repr__())]]
+        for t in ngmesh.Elements2D():
+            T = T + [[int(v.__repr__())-1 for v in t.vertices]]
```

Comment:
I didn't know I could have used `v.nr`. I've implemented this suggestion. A possible way of avoiding the loop over all `Elements2D` could be to expose using pybind11 `Mesh::SurfaceElements`, I'll fork the Netgen repo and see what I can do :)

Reply:
Ok, now it should be a bit better, but in all fairness, I just moved the loops on the C++ side of the Netgen wrapper. This seems to be speeding up things significantly, but probably the best way to do it would be to use the memoryview feature from Pybind11.